balancer-js/src/modules/join/join.module.integration.spec.ts

```diff
@@ -0,0 +1,347 @@
+import dotenv from 'dotenv';
+import { expect } from 'chai';
+import { BalancerSDK, Network, PoolModel, PoolToken } from '@/.';
+import hardhat from 'hardhat';
+
+import { JsonRpcSigner, TransactionReceipt } from '@ethersproject/providers';
+import { BigNumber } from 'ethers';
+import { parseFixed } from '@ethersproject/bignumber';
+
+import { balancerVault } from '@/lib/constants/config';
+import { AddressZero } from '@ethersproject/constants';
+
+dotenv.config();
+
+const { ALCHEMY_URL: jsonRpcUrl } = process.env;
+const { ethers } = hardhat;
+
+let balancer: BalancerSDK;
+const rpcUrl = 'http://127.0.0.1:8545';
+const provider = new ethers.providers.JsonRpcProvider(rpcUrl, 1);
+const signer = provider.getSigner();
+let signerAddress: string;
+
+// Slots used to set the account balance for each token through hardhat_setStorageAt
+// Info fetched using npm package slot20
+const wETH_SLOT = 3;
+const wBTC_SLOT = 0;
+
+const initialBalance = '100000';
+const amountsInDiv = '100000000';
+
+let tokensIn: PoolToken[];
+let amountsIn: string[];
+
+// Setup
+
+const setupPool = async () => {
```

Comment:
We'll probably use a lot of these functions in other tests so lets add them to a utils in test/lib.

Reply:
I moved what I considered to be reusable. Let me know if you expect to see other functions there as well.